File: api/src/main/java/work/beltran/discogsbrowser/api/model/record/BasicInformation.java
package work.beltran.discogsbrowser.api.model.record;

import com.google.auto.value.AutoValue;
import com.google.gson.Gson;
import com.google.gson.TypeAdapter;

import java.util.List;

/**
 * Created by Miquel Beltran on 23.04.16.
 * More on http://beltran.work
 */
@AutoValue
public abstract class BasicInformation {
    public abstract String getTitle();
    public abstract String getYear();
    public abstract String getThumb();
    public abstract List<Format> getFormats();
    public abstract List<Artist> getArtists();

    public static TypeAdapter<BasicInformation> typeAdapter(Gson gson) {
        return new AutoValue_BasicInformation.GsonTypeAdapter(gson);
    }
    public static Builder builder() {
        return new AutoValue_BasicInformation.Builder();
    }

    @AutoValue.Builder
    public abstract static class Builder {
        public abstract Builder title(String title);
        public abstract Builder year(String year);
        public abstract Builder thumb(String thumb);
        public abstract Builder formats(List<Format> formats);
        public abstract Builder artists(List<Artist> artists);
        public abstract BasicInformation build();
    }
}

File: api/src/main/java/work/beltran/discogsbrowser/api/model/record/BasicInformation.kt
package work.beltran.discogsbrowser.api.model.record

/**
 * Created by Miquel Beltran on 23.04.16.
 * More on http://beltran.work
 */
data class BasicInformation(
        val title: String,
        val year: String,
        val thumb: String,
        val formats: List<Format>,
        val artists: List<Artist>
)
